### PR TITLE
To train on 3DMatch with pointnet++

### PR DESCRIPTION
**Pull request by humanpose1**
I've corrected some mistakes and I have an example of training using a pointnet++ like model. There are still some problems of convergence (metric learning on patches is hard...) but now we can launch the training.
```
python train.py task=registration model_type=pointnet2 model_name=minipointnet2 dataset=patch3dmatch training=default_reg lr_scheduler=multi_step_reg
```

### CODE REVIEW

File: conf/lr_scheduler/multi_step_reg.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+lr_scheduler:
+  class: MultiStepLR
+  params:
+    milestones: [3000, 6000, 9000, 15000, 21000, 30000]
```
Comment (tchaton) [inline conf/lr_scheduler/multi_step_reg.yaml]: The BaseModel is providing epoch and not steps anymore. Those won t be reached

File: src/datasets/registration/pair.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import torch
+from torch_geometric.data import Data
+from torch_geometric.data import Batch
+
+
+def make_pair(data_source: Data, data_target: Data):
```
Comment (tchaton) [inline src/datasets/registration/pair.py]: Maybe this one should be moved to some datasets utile along with others batch aggregations

File: src/core/data_transform/features.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import torch
+
+
+class OneFeature(object):
```
Comment (tchaton) [inline src/core/data_transform/features.py]: Why do u need this ? If it is KPConv, I will add this within a PR on Monday ?

File: src/core/data_transform/features.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        assert data.pos is not None
+        xyz = data.pos[:, self.axis]
+        if data.x is None:
+            data.x = xyz
+        else:
+            data.x = torch.cat([data.x, xyz], -1)
+        return data
+
+
+class RGBFeature(object):
```
Comment (tchaton) [inline src/core/data_transform/features.py]: Add normalize option to put rgb within 0, 1

**Comment by tchaton** [comment]
@humanpose1, awesome PR ! I will give it a try :)

File: src/core/data_transform/features.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        return data
+
+
+class NormalFeature(object):
+    """
+    add normal as feature. if it doesn't exist, compute normals
+    using PCA
+    """
+    def __call__(self, data):
+        if data.norm is None:
```
Comment (nicolas-chaulet) [inline src/core/data_transform/features.py]: rename to data.normal

Comment (humanpose1) [reply src/core/data_transform/features.py]: As you want but, I use norm because in torch_geometric they use also that [name.](https://github.com/rusty1s/pytorch_geometric/blob/18da46c2a28faad83aca9fb7602500c69fc2d4b0/torch_geometric/data/data.py#L47)

Comment (nicolas-chaulet) [reply src/core/data_transform/features.py]: Alright, leave it as is then! I did not know that
